aten/src/ATen/test/complex_test.cpp

```diff
@@ -0,0 +1,34 @@
+#include <gtest/gtest.h>
+#include "c10/util/Complex.h"
+
+template<typename T, typename iT>
+static void TestIntegerOp(T real, T img, iT i) {
+  std::complex<T> c(real, img);
+  ASSERT_EQ(c + i, std::complex<T>(real + i, img + i));
+  ASSERT_EQ(i + c, std::complex<T>(i + real, i + img));
+  ASSERT_EQ(c - i, std::complex<T>(real - i, img - i));
+  ASSERT_EQ(i - c, std::complex<T>(i - real, i - img));
+  ASSERT_EQ(c * i, std::complex<T>(real * i, img * i));
+  ASSERT_EQ(i * c, std::complex<T>(i * real, i * img));
+  ASSERT_EQ(c / i, std::complex<T>(real / i, img / i));
+  ASSERT_EQ(i / c, std::complex<T>(i / real, i / img));
+}
+
+template<typename Op>
+static void TestIntegerOpAllTypes(float real, float img, int8_t i) {
+  TestIntegerOp<float, int8_t>(real, img, i, op);
+  TestIntegerOp<double, int8_t>(real, img, i, op);
+  TestIntegerOp<float, int16_t>(real, img, i, op);
+  TestIntegerOp<double, int16_t>(real, img, i, op);
+  TestIntegerOp<float, int32_t>(real, img, i, op);
+  TestIntegerOp<double, int32_t>(real, img, i, op);
+  TestIntegerOp<float, int64_t>(real, img, i, op);
+  TestIntegerOp<double, int64_t>(real, img, i, op);
+}
+
+TEST(ComplexTest, Integer) {
+  TestIntegerOpAllTypes(1.0, 0.1, 1);
```

Comment:
I think one call instead of four should be fine and you could instead have just calls like:
```
TestIntegerOpAllTypes<float>(1.2, -0.2, -2);
TestIntegerOpAllTypes<double>(1.2, -0.2, -2);
```

Reply:
Yeah that could work, but I feel having four calls on different combinations of signs may prevent introducing some issues in the future.

Reply:
(Since we have four calls, I did not separate `float` and `double` to two different calls)

Reply:
you could half the number of calls by templatizing it because you have similar calls for float and half.

Reply:
yeah but the four cases do not test any additional things other than the fact that complex number is being stored properly. you can add a separate test for that if you want

Reply:
OK, I removed the additional calls.